pandas/_libs/reduction.pyx

```diff
@@ -24,9 +24,9 @@ is_numpy_prior_1_6_2 = LooseVersion(np.__version__) < '1.6.2'
 
 cdef _get_result_array(object obj, Py_ssize_t size, Py_ssize_t cnt):
 
-    if isinstance(obj, np.ndarray) \
-            or isinstance(obj, list) and len(obj) == cnt \
-            or getattr(obj, 'shape', None) == (cnt,):
+    if (util.is_array(obj) or
+            isinstance(obj, list) and len(obj) == cnt or
```

Comment:
could do PyListCheck here

Reply:
or change the other below, I find this slightly non-idiomatic

Reply:
It looks like cython automatically converts to `PyList_Check`.

Agreed on a lot of the idiomatic point.  There's a lot of built-up debris in _libs.